src/io/flutter/devtools/DevToolsUtils.java

```diff
@@ -54,4 +54,15 @@ public static String generateDevToolsUrl(
     }
     return "http://" + devtoolsHost + ":" + devtoolsPort + "/?" + String.join("&", params);
   }
+
+  public static String findWidgetId(String url) {
+    final String searchFor = "inspectorRef=";
+    final String[] split = url.split("&");
+    for (String part : split) {
+      if (part.startsWith(searchFor)) {
+        return part.substring(searchFor.length());
+      }
+    }
+    return null;
```

Comment:
instead of doing a string search, should we do URI parsing instead and then grab the query param value at 'inspectorRef'?

Reply:
Weirdly I can't seem to get a query string from the URL. e.g. 
```
final URI uri = new URI(urlString);
final String[] paramPairs = uri.getQuery() // this is null
```

maybe because of the `#` after the port? I'm not sure. Certainly it would be nicer to use more standard parsing, if anyone has suggestions on this.